cmake/DaemonBuiltin.cmake

```diff
@@ -0,0 +1,29 @@
+set(DAEMON_BUILTIN_HEADER "// Automatically generated, do not modify!\n")
+set(DAEMON_BUILTIN_C "${DAEMON_BUILTIN_HEADER}")
+set(DAEMON_BUILTIN_H "${DAEMON_BUILTIN_HEADER}")
+
+macro(daemon_add_builtin TYPE NAME VALUE)
+	set(DAEMON_BUILTIN_C "${DAEMON_BUILTIN_C}const ${TYPE} ${NAME}=${VALUE};\n")
+	set(DAEMON_BUILTIN_H "${DAEMON_BUILTIN_H}extern const ${TYPE} ${NAME};\n")
+endmacro()
+
+set(DAEMON_BUILTIN_DIR "${CMAKE_CURRENT_BINARY_DIR}/builtin")
```

Comment:
It might be nice if `builtin.cpp` etc. were not a singleton... so we could have a Daemon one and an Unvanquished one with other information e.g. an Unvanquished git revision. Non-blocking comment.

Reply:
Yes but that would make it more complicated. One good news is that VM buildt as subproject already use their own `builtin.cpp` file. We can split it later if we face the need for it.